[SID-1206] Add renderLabel prop to <OrganizationSwitcher>

jake-slashid opened the pull request:
## Description

[YouTrack issue](https://todo.irdcorp.dev/issue/SID-1206)

Two changes
- Add `renderLabel` prop to `<OrganizationSwitcher>`, see context.
    - I'm changing `Dropdown` to allow `ReactNode` for `label`
    - The type is now `{ value: string, label: ReactNode }`
- Add npm script `next`. It automatically publishes things to `yalc` with a bumped pre-release version.

### Context
The org list currently renders org names "as is" which can create some interesting problems for consumers when weighing up the options for choosing a programmatically generated organization name scheme.

This change proposes allowing org name overrides (i.e. display names) in the most prominent place the org name is shown: the org switcher.

This feature gives flexibility to consumers to choose internal-only names for programmatically generated organizations, or even to implement internationalisation on org names if they wish.

```tsx
<OrganizationSwitcher
  renderLabel={org => {
    if (org.id === "some_id") return "Some custom org name!"
    return org.org_name
  }}
/>
```

```tsx
<OrganizationSwitcher
  renderLabel={org => {
    return i18n(org.id)
  }}
/>
```

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have generated the new version of the docs website and smoke tested it
- [x] I have checked that my changes haven't caused semantic errors in the existing docs
- [x] I have updated the README and DEVELOPMENT files

jake-slashid commented on packages/react/src/components/organization-switcher/index.tsx:
```diff
@@ -10,6 +10,7 @@ import { OrganizationDetails } from "@slashid/slashid";
 interface Props {
   fallback?: ReactNode;
   filter?: (organization: OrganizationDetails) => boolean;
+  renderLabel?: (organization: OrganizationDetails) => ReactNode;
```
I intentionally did not name this `renderItem` because it does not allow you to control the full rendering of the menu item only the label.

The selected item for example has a checkmark, we don't allow you to configure this (yet).

I anticipate in future we would also have `renderItem` which would allow control of the entire item renderer, but out of scope for now.

jake-slashid reviewed (commented):
Here is a practical example of how this feature is useful:

In this multi-tenancy demo I can present "Your personal organization" to every user, greatly simplifying the UX

![image](https://github.com/slashid/javascript/assets/139138253/35595e0b-189e-4f73-a191-fa6c62d2914c)

kaspermroz commented on packages/react/package.json:
```diff
@@ -43,7 +43,8 @@
     "test:unit": "vitest run",
-    "test:unit:watch": "vitest"
+    "test:unit:watch": "vitest",
+    "next": "node ./yalc.next.js && pnpm build && yalc publish"
```
I'd think of a slightly more meaningful name for it, `next` makes me think of Next.js. How about `next-version` or `prerelease` ?

kaspermroz replied:
or even `yalc` / `yalc-publish`

ikovic replied:
Following the conventions we had so far, my vote would be `publish:local` or `publish:yalc`

jake-slashid replied:
I renamed it `publish:yalc`

ikovic commented:
@jake-slashid the deployment on cloudflare complains about a broken lockfile but it looks good to me 🤔  it might be using another version of PNPM though

jake-slashid commented:
@ikovic the lockfile was broken when I first pushed, it's green now.

Does it auto deploy on push or can we trigger it again somehow?

ikovic commented:
> @ikovic the lockfile was broken when I first pushed, it's green now.
> 
> Does it auto deploy on push or can we trigger it again somehow?

Last build was executed on top of [a96d101](https://github.com/slashid/javascript/pull/136/commits/a96d10126d24f7cfe95497fe266f619506497f0a). I'll retry the deployment from the Cloudflare dashboard

jake-slashid commented:
Docs: https://github.com/slashid/ng-evangelion/pull/995

jake-slashid commented:
Ready for final review!

- Added storybook example
- Added tests
- Wrote docs: https://github.com/slashid/ng-evangelion/pull/995
- Addressed PR feedback
- CI green ✅

kaspermroz reviewed (approved):
Just couple minor comments, other than that it looks good! 🎉

jake-slashid commented on .changeset/cyan-mangos-judge.md:
```diff
@@ -0,0 +1,5 @@
+"@slashid/react": minor
+---
+
+Add renderLabel prop to \<OrganizationSwitcher />
```
It has to be here because github renders HTML 

Here's an example: \<Hello />
Here's an example: <Hello />

jake-slashid commented on packages/react/src/components/organization-switcher/organization-switcher.test.tsx:
```diff
@@ -139,6 +139,38 @@ describe("OrganizationSwitcher", () => {
+    );
+
+    await waitFor(() =>
+      expect(screen.queryByText("-")).not.toBeInTheDocument()
```
You're right but I couldn't think of another way to do it, because the loading state is actually the same component as the rendered state - there are no differences.

I think testing for the current value is probably a better idea, let me try

jake-slashid replied:
Done! It's a bit noisy but more future proof
